### PR TITLE
List known `xref` props with description and regex

### PR DESCRIPTION
**Pull request by generalmimon**
Descriptions are taken from the [User Guide](https://doc.kaitai.io/user_guide.html#meta-xref), only shortened. Some regexes are hand-crafted, others I took from Wikidata (I always add some description of the regex to the `$comment` key).

### CODE REVIEW

File: ksy_schema.json [hunk 1]
```diff
@@ -82,6 +82,151 @@
         "xref": {
           "type": "object",
           "additionalProperties": false,
+          "properties": {
+            "forensicswiki": {
+              "description": "article name at [Forensics Wiki](https://forensicswiki.org/wiki/Main_Page), which is a CC-BY-SA-licensed wiki with information on digital forensics, file formats and tools\n\nfull link name could be generated as `https://forensicswiki.org/wiki/` + this value",
```
Comment (dgelessus) [inline ksy_schema.json]: This should be fixed in the original documentation as well, but https://forensicswiki.org/ went down a few months ago, and the site is apparently located at https://forensicswiki.xyz/ now.

File: ksy_schema.json [hunk 2]
```diff
@@ -82,6 +82,151 @@
...
+              "anyOf": [
+                {
+                  "type": "string",
+                  "pattern": "^([a-zA-Z0-9$\\-._~+!*'(),@&;:\\/]+|%[0-9a-fA-F]{2})+$",
+                  "$comment": "hand-crafted regex, it should match any valid URL path"
+                },
```
Comment (dgelessus) [inline ksy_schema.json]: Might be a good idea to extract this into a separate definition (e. g. `MediaWikiPageName`), as we have multiple `xref` keys of this kind (`forensicswiki` and `justsolve`).

**Review by dgelessus — COMMENTED** [review]
Good idea! I have a couple of minor comments.

Also a more general comment - for each of the `xref` keys, the value type is currently listed twice (once for a single scalar value, and once for an array). I'm not a big fan of duplicating information like that, as it's easy to update only one of the two definitions by accident. On the other hand, I'm not sure if there's a much better solution - you could extract each type into a definition, but that might not be worth the effort.

**Comment by generalmimon** [comment]
I forgot to add: here is the [**documentation generated by Docson from this schema**](https://lbovet.github.io/docson/index.html#https://raw.githubusercontent.com/kaitai-io/ksy_schema/f5bd10fdf8/ksy_schema.json) (please be patient, takes a long time before it loads: 15-20 seconds). I intend to replace the KSY reference with the documentation from Docson entirely (as I suggest in https://github.com/kaitai-io/kaitai_struct/issues/655#issuecomment-614299673), so I'm gradually moving the information towards KSY schema. With Markdown syntax that Docson supports, it's quite easy to achieve even advanced formatting, see the table I created at `params/ParamsSpec/ParamSpec/type` in the demo I refer to.

@dgelessus Thanks for reviewing!

> Also a more general comment - for each of the `xref` keys, the value type is currently listed twice (once for a single scalar value, and once for an array). I'm not a big fan of duplicating information like that, as it's easy to update only one of the two definitions by accident.

A valid concern, I completely agree with this. I was a bit unsure about creating a special definition for each pattern just to reuse it on two places, but it definitely has a maintainability benefit. I think it's worth the effort.

**Review by dgelessus — APPROVED** [review]

File: ksy_schema.json (hunk 1 above)
Comment (generalmimon) [reply ksy_schema.json]: @dgelessus Thanks for the info, I didn't know that. I'll fix it.

**Comment by generalmimon** [comment]
@dgelessus Thanks for your review! I seem to have incorporated all suggestions, so I'm merging this PR.

**Pull request merged**